extensions/blocks/map/map.php

```diff
@@ -40,6 +40,22 @@ function jetpack_get_mapbox_api_key() {
 	return Jetpack_Options::get_option( 'mapbox_api_key' );
 }
 
+/**
+ * Record a Tracks event every time the Map block is loaded on WordPress.com and Atomic.
+ */
+function jetpack_record_mapbox_load_event() {
+	$event_name = 'jetpack_map_block_mapbox_load';
+	$user       = wp_get_current_user();
+	if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+		require_lib( 'tracks/client' );
+		tracks_record_event( $user, $event_name, array( 'site_id' => get_current_blog_id() ) );
+		return;
+	} elseif ( jetpack_is_atomic_site() ) {
+		$tracking = new Automattic\Jetpack\Tracking();
+		$tracking->tracks_record_event( $user, $event_name, array( 'site_id' => Jetpack_Options::get_option( 'id' ) ) );
```

Comment:
Maybe you can use `record_user_event` here in Jetpack? It offers the advantage of taking care of adding the blog ID (and more) for you:
https://github.com/Automattic/jetpack/blob/1b981c3573c82a4c83b3c212a241b59097691792/packages/tracking/src/class-tracking.php#L60-L77

Reply:
Since we need the site ID to do usage checks, I'd rather explicitly add it as custom event property and have full control over it.
It's obviously very unlikely, but what if at some point we decide to stop recording the `blog_id`, or change its name?

(Also, but this might very well be my fault: on Tracks the global property is called `blogid` and I've seen that it often returns `NULL`... Better safe than sorry I guess?)

Reply:
> Since we need the site ID to do usage checks, I'd rather explicitly add it as custom event property and have full control over it.

Since you're pulling it from the same place as `record_user_event` does, you should always have similar results.

> what if at some point we decide to stop recording the blog_id, or change its name?

That's unlikely, as it's a key piece on both Jetpack and wpcom.

If you want to be double sure you could add an extra `site_id` property to the data you pass to `record_user_event`, so you'd have it twice if you'd like. It does seem a little bit overkill though.

> this might very well be my fault: on Tracks the global property is called blogid and I've seen that it often returns NULL... Better safe than sorry I guess?

That is odd indeed, and may be worth its own investigation.

Reply:
Might very well be because people are using `tracks_record_event` instead of `record_user_event` 😄